File: MAG/MAG.Model/Anime.cs
namespace MAG.Model
{
    public class Anime
    {
        public int Id { get; set; }

        public string TitleEn { get; set; } = null!;

        public string TitleJp { get; set; } = null!;

        public string Synopsis { get; set; } = null!;

        public int EpisodesNumber { get; set; }

        public string? ImageUrl { get; set; }

        public string? TrailerUrl { get; set; }

        public double Score { get; set; }

        public DateTime BeginAir { get; set; }

        public DateTime FinishAir { get; set; }

        public string Season { get; set; } = null!;

        public string Studio { get; set; } = null!;

        public virtual ICollection<GenreAnime> GenreAnimes { get; set; } = new List<GenreAnime>();

    }
}

File: MAG/UI/mag_user/lib/models/anime.dart
import 'package:json_annotation/json_annotation.dart';
import '../models/genre_anime.dart';
part 'anime.g.dart';

// flutter pub run build_runner build --delete-conflicting-outputs -> one time
// flutter pub run build_runner watch --delete-conflicting-outputs -> continuous

@JsonSerializable()
class Anime {
  int? id;
  String? titleEn;
  String? titleJp;
  String? synopsis;
  int? episodesNumber;
  String? imageUrl;
  String? trailerUrl;
  double? score;
  DateTime? beginAir;
  DateTime? finishAir;
  String? season;
  String? studio;
  List<GenreAnime>? genreAnimes;
  Anime(
    this.id,
    this.titleEn,
    this.titleJp,
    this.synopsis,
    this.episodesNumber,
    this.imageUrl,
    this.trailerUrl,
    this.score,
    this.beginAir,
    this.finishAir,
    this.season,
    this.studio,
    this.genreAnimes,
  );

  factory Anime.fromJson(Map<String, dynamic> json) => _$AnimeFromJson(json);

  Map<String, dynamic> toJson() => _$AnimeToJson(this);
}
